src/plugins/vis_type_timelion/public/components/timelion_vis_component.tsx

```diff
@@ -378,32 +143,77 @@ function TimelionVisComponent({
     [fireEvent]
   );
 
-  useEffect(() => {
-    if (chartElem) {
-      $(chartElem).off('plotselected').on('plotselected', plotSelectedHandler);
-    }
-  }, [chartElem, plotSelectedHandler]);
-
-  useEffect(() => {
-    if (chartElem) {
-      $(chartElem).off('mouseleave').on('mouseleave', mouseLeaveHandler);
-    }
-  }, [chartElem, mouseLeaveHandler]);
-
-  useEffect(() => {
-    if (chartElem) {
-      $(chartElem).off('plothover').on('plothover', plotHoverHandler);
-    }
-  }, [chartElem, plotHoverHandler]);
+  const onRenderChange = useCallback(
+    (isRendered: boolean) => {
+      if (!isRendered) {
+        renderComplete();
+      }
+    },
+    [renderComplete]
+  );
 
   const title: string = useMemo(() => last(compact(map(seriesList.list, '_title'))) || '', [
     seriesList.list,
   ]);
 
   return (
-    <div ref={elementRef} className="timChart">
-      <div className="chart-top-title">{title}</div>
-      <div ref={canvasRef} className="chart-canvas" />
+    <div className="timelionChart">
+      <div className="timelionChart__topTitle">{title}</div>
+      <Chart renderer="canvas" size={{ width: '100%' }}>
+        <Settings
+          onBrushEnd={brushEndListener}
+          showLegend
+          legendPosition={getLegendPosition()}
+          onRenderChange={onRenderChange}
+          theme={kibana.services.chartTheme.useChartsTheme()}
```

Comment:
You can see above Stratoula's comment about problems with bars and crosshair line. I talked with Marco about this. He said that we should use elastic-chart themes instead of hardcode 'red' color for crosshair line. In Elastic-chart themes it's light gray color for crosshair (I think it will be less terrible with bars). Also he explained that when we use 'BarSeries' chart crosshair line adapts by bucket interval because it's not logical to check any time when we use 'bars'. It's a not an area or a line where we can check any time, in 'BarSeries' the value doesn't change within the specified interval because of this we move form one interval to another. In this case red crosshair line looks not good for 'BarSeries'. I think we can somehow set red line only for Area and Line if it needed or choose a new color which preferable for all types of chart in timelion. @markov00, @stratoula what do you think?

Reply:
I agree with Marco, let's use the elastic charts crosshair line. Now about the line on bars I am more positive to only allow it for areas and lines and not bars but I would love @markov00 's opinion on that too

Reply:
Sorry for being late here: yes IMHO we should align the style across all the charts using the Kibana ECH theme provided. If we want to change the color for that cursor, we can do it one time for every chart in a single place